src/global.cpp

```diff
@@ -88,6 +93,61 @@ std::shared_future<void> Cleanup() { return impl::Init::Instance().cleanup(); }
 
 void SetSctpSettings(SctpSettings s) { impl::Init::Instance().setSctpSettings(std::move(s)); }
 
+std::map<int, UnhandledStunRequestHandler *> unboundStunCallbacks;
```

Comment:
Should callbacks be specified by port or by bind address? Just asking because it needs to be consistent with libjuice. I think port makes sense, since the user is typically going to set the same `bindAddress` everytime, and the any address is quite tedious to handle.

Reply:
The [libjuice PR](https://github.com/paullouisageneau/libjuice/pull/292) uses `${host}:${port}` as the key.

Since the `conn_mux.c` uses `udp_socket_config_t` to bind a UDP port, I wanted to make sure the user could explicitly bind both IPv4 and IPv6 addresses with:

```c
juice_mux_listen("0.0.0.0", port, &cb, NULL);
juice_mux_listen("::", port, &cb, NULL);
```

I could be misreading it but I think [these lines](https://github.com/paullouisageneau/libjuice/blob/master/src/udp.c#L161-L172) mean you need to pass one address family or the other.  For example passing `NULL` as the `bind_address` will not cause it to bind to both.

Reply:
> Actually I'm not sure that's correct, passing `NULL` seems to bind all IPv4 and IPv6 interfaces.

Correct, `NULL` binds on IPv6 and IPv4. That's what you want in most cases.